app/services/mapping/declaration/FunctionalReferenceIdBuilder.scala

```diff
@@ -33,6 +35,16 @@ object FunctionalReferenceIdBuilder {
     referenceId
   }
 
+  def buildThenAdd(exportsCacheModel: ExportsCacheModel, declaration: Declaration) {
+    exportsCacheModel.consignmentReferences.foreach(references => {
+      val declarationFunctionalReferenceIdType = new DeclarationFunctionalReferenceIDType()
```

Comment:
I would move the instantiation inside the if statement. 
We only want to create an instance if we have an `LRN`

Reply:
yep agreed, have changed.